test/unit-test/dawn/Optimizer/Passes/TestTemporaryToFunction.cpp

```diff
@@ -0,0 +1,123 @@
+//===--------------------------------------------------------------------------------*- C++ -*-===//
+//                          _
+//                         | |
+//                       __| | __ ___      ___ ___
+//                      / _` |/ _` \ \ /\ / / '_  |
+//                     | (_| | (_| |\ V  V /| | | |
+//                      \__,_|\__,_| \_/\_/ |_| |_| - Compiler Toolchain
+//
+//
+//  This file is distributed under the MIT License (MIT).
+//  See LICENSE.txt for details.
+//
+//===------------------------------------------------------------------------------------------===//
+
+#include "dawn/CodeGen/CXXNaive/CXXNaiveCodeGen.h"
+#include "dawn/CodeGen/CodeGen.h"
+#include "dawn/CodeGen/GridTools/GTCodeGen.h"
+#include "dawn/Compiler/DawnCompiler.h"
+#include "dawn/Compiler/DawnCompiler.h"
+#include "dawn/Compiler/Options.h"
+#include "dawn/Optimizer/OptimizerContext.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/SIR/SIR.h"
+#include "dawn/SIR/SIRSerializer.h"
+#include "dawn/Support/EditDistance.h"
+#include "dawn/Support/Logging.h"
+#include "dawn/Support/StringSwitch.h"
+#include "dawn/Support/StringUtil.h"
+#include "dawn/Support/Unreachable.h"
+#include "test/unit-test/dawn/Optimizer/Passes/TestEnvironment.h"
+#include <fstream>
+#include <gtest/gtest.h>
+#include <streambuf>
+
+#include "dawn/Optimizer/PassComputeStageExtents.h"
+#include "dawn/Optimizer/PassDataLocalityMetric.h"
+#include "dawn/Optimizer/PassFieldVersioning.h"
+#include "dawn/Optimizer/PassInlining.h"
+#include "dawn/Optimizer/PassMultiStageSplitter.h"
+#include "dawn/Optimizer/PassPrintStencilGraph.h"
+#include "dawn/Optimizer/PassSSA.h"
+#include "dawn/Optimizer/PassSetCaches.h"
+#include "dawn/Optimizer/PassSetNonTempCaches.h"
+#include "dawn/Optimizer/PassSetStageGraph.h"
+#include "dawn/Optimizer/PassSetStageName.h"
+#include "dawn/Optimizer/PassStageMerger.h"
+#include "dawn/Optimizer/PassStageReordering.h"
+#include "dawn/Optimizer/PassStageSplitter.h"
+#include "dawn/Optimizer/PassStencilSplitter.h"
+#include "dawn/Optimizer/PassTemporaryFirstAccess.h"
+#include "dawn/Optimizer/PassTemporaryMerger.h"
+#include "dawn/Optimizer/PassTemporaryToStencilFunction.h"
+#include "dawn/Optimizer/PassTemporaryType.h"
+
+using namespace dawn;
+
+namespace {
+
+class TemporaryToFunction : public ::testing::Test {
+
+  dawn::DawnCompiler compiler_;
+
+protected:
+  TemporaryToFunction() {
+    compiler_.getOptions().PassTmpToFunction = true;
+    compiler_.getOptions().ReportPassTmpToFunction = true;
```

Comment:
why do we want the report enabled in this test? we don't use the cout output anywhere

Reply:
gone